app/controllers/shelters_controller.rb

```diff
@@ -12,28 +12,29 @@ def new
   end
 
   def create
-    if Shelter.create(shelter_params).valid?
+     shelter = Shelter.create(shelter_params)
+    if shelter.valid?
       flash[:notice] = "You've successfully created a shelter."
       redirect_to '/shelters'
-    else   
-      flash[:notice] = "The following fields are missing : #{Shelter.empty_params(shelter_params)}"
+    else
+      flash[:error] = "#{shelter.errors.full_messages.to_sentence}"
       redirect_to '/shelters/new'
     end
   end
 
   def edit
     shelter
   end
-  
+
   def update
-    # shelter.update(shelter_params)
-    # redirect_to "/shelters/#{shelter.id}"
-    if shelter.update(shelter_params)
+    shelter_1 = shelter
```

Comment:
calling the `shelter` helper method to access the `@shelter` instance variable did not produce the error sentence we need for the flash message